packages/app-next/src/App.tsx

```diff
@@ -76,6 +77,9 @@ const app = createApp({
   ],
   bindRoutes({ bind }) {
     bind(pagesPlugin.externalRoutes, { pageX: pagesPlugin.routes.pageX });
+    bind(catalogPlugin.externalRoutes, {
+      viewTechDoc: techdocsPlugin.routes.docRoot,
```

Comment:
Created a separate pull request for exporting the TechDocs routes: https://github.com/backstage/backstage/pull/20605/files#diff-c34ee22bb6f58cec5c50bd549823f9e998e98aab7e3615abda60f22c3971e0c6R164